pytradfri/api/libcoap_api.py

```diff
@@ -21,25 +22,34 @@
 class APIFactory:
     """APIFactory."""
 
-    def __init__(self, host, psk_id="pytradfri", psk=None, timeout=10):
+    def __init__(
+        self,
+        host: str,
+        psk_id: str = "pytradfri",
+        psk: str | None = None,
+        timeout: int = 10,
+    ) -> None:
         """Create object of class."""
         self._host = host
         self._psk_id = psk_id
         self._psk = psk
         self._timeout = timeout  # seconds
 
     @property
-    def psk(self):
+    def psk(self) -> str | None:
         """Return psk."""
         return self._psk
 
     @psk.setter
-    def psk(self, value):
+    def psk(self, value: str) -> None:
         """Set psk."""
         self._psk = value
 
     def _base_command(self, method: str) -> list[str]:
         """Return base command."""
+        if self._psk is None:
```

Comment:
Not sure about this one, but `_psk` will not always be a string so we have to handle that somehow.

Reply:
Raise a RuntimeError. The user needs to generate a key first.